solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -636,6 +651,8 @@ public void load() {
     updateShardHandler = new UpdateShardHandler(cfg.getUpdateShardHandlerConfig());
     updateShardHandler.initializeMetrics(solrMetricsContext, "updateShardHandler");
 
+    solrClientCache = new SolrClientCache(updateShardHandler.getDefaultHttpClient());
```

Comment:
Maybe this is an aside by why is `UpdateShardHandler` called what it is?  Is it just for "updates"?  It has no javadocs.  It seems a little weird in this line to get an httpclient from an "update" thingy here -- seems arbitrary.  If UpdateShardHandler was perhaps named something more general and not specific to updates, I wouldn't be posing this question.

Reply:
This is a good question but it's out of scope for this issue.

This usage comes straight from refactoring - currently whenever the code needs a new SolrClient it uses the HttpClient from UpdateShardHandler.